terraform/bod_private_dns.tf

```diff
@@ -0,0 +1,153 @@
+resource "aws_route53_zone" "bod_private_zone" {
+  name = "${local.bod_private_domain}."
+  vpc_id = "${aws_vpc.bod_vpc.id}"
+  tags = "${merge(var.tags, map("Name", "BOD Private Zone"))}"
+  comment = "Terraform Workspace: ${lookup(var.tags, "Workspace", "Undefined")}"
+}
+
+resource "aws_route53_record" "bod_router_A" {
+  zone_id = "${aws_route53_zone.bod_private_zone.zone_id}"
+  name = "router.${aws_route53_zone.bod_private_zone.name}"
+  type = "A"
+  ttl = 300
+  records = [
+    "${cidrhost(aws_subnet.bod_public_subnet.cidr_block, 1)}",
+    "${cidrhost(aws_subnet.bod_private_subnet.cidr_block, 1)}"
+  ]
+}
+
+resource "aws_route53_record" "bod_ns_A" {
+  zone_id = "${aws_route53_zone.bod_private_zone.zone_id}"
+  name = "ns.${aws_route53_zone.bod_private_zone.name}"
+  type = "A"
+  ttl = 300
+  records = [
+    "${cidrhost(aws_subnet.bod_public_subnet.cidr_block, 2)}",
+    "${cidrhost(aws_subnet.bod_private_subnet.cidr_block, 2)}"
+  ]
+}
+
+resource "aws_route53_record" "bod_reserved_A" {
+  zone_id = "${aws_route53_zone.bod_private_zone.zone_id}"
+  name = "reserved.${aws_route53_zone.bod_private_zone.name}"
+  type = "A"
+  ttl = 300
+  records = [
+    "${cidrhost(aws_subnet.bod_public_subnet.cidr_block, 3)}",
+    "${cidrhost(aws_subnet.bod_private_subnet.cidr_block, 3)}"
+  ]
+}
+
+resource "aws_route53_record" "bod_bastion_A" {
+  zone_id = "${aws_route53_zone.bod_private_zone.zone_id}"
+  name = "bastion.${aws_route53_zone.bod_private_zone.name}"
+  type = "A"
+  ttl = 300
+  records = [
+    "${aws_instance.bod_bastion.public_ip}"
```

Comment:
Should this be the private IP?  I copied what was done in the CyHy VPC, but this doesn't seem correct.

Reply:
@felddy, care to comment about this?

Reply:
I'm trying to find where @jsf9k  copied this from.  The private bastion entries I wrote don't reference the `public_ip`:
https://github.com/dhs-ncats/cyhy_amis/blob/bb304156750f0028b453ff6bef94bca92b9da700/terraform/cyhy_private_dns.tf#L35

Also, the entries in route53 look correct in production.

Reply:
You are correct sir. I don't know where I got that from.